core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -669,4 +724,45 @@ public static String testBlobPrefix(String seed) {
     public boolean readOnly() {
         return readOnly;
     }
+
+    @Override
+    public List<SnapshotId> resolveSnapshots(final String... snapshotNames) {
+        List<SnapshotId> ids = new ArrayList<>();
+        final List<SnapshotId> snapshotIds = snapshots();
+        for (String snapshotName : snapshotNames) {
+            SnapshotId resolvedSnapshotId = null;
+            for (SnapshotId snapshotId : snapshotIds) {
+                if (snapshotName.equals(snapshotId.getName())) {
+                    resolvedSnapshotId = snapshotId;
+                }
+            }
+            if (resolvedSnapshotId == null) {
+                throw new SnapshotMissingException(new Snapshot(repositoryName, snapshotName));
+            }
+            ids.add(resolvedSnapshotId);
+        }
+        return Collections.unmodifiableList(ids);
+    }
+
+    // TODO: this will go away once readSnapshotsList uses the index file instead of listing blobs
+    // to know which snapshots are part of a repository.  See #18156
+    private Tuple<String, String> parseNameUUIDFromBlobName(final String str) {
```

Comment:
Its a good idea, though this method is disappearing in the very next PR (I already have it implemented), where we do not depend on blob listing and parsing the UUID from the blob name.  So perhaps thoroughly testing this method in particular won't be a good use of time as it will disappear very soon thereafter.  Also, the other tests have captured issues with parsing the blob names that I've uncovered and fixed in this method.  So its not entirely untested.